include/odp/api/spec/packet.h

```diff
@@ -1140,6 +1179,82 @@ int odp_packet_move_data(odp_packet_t pkt, uint32_t dst_offset,
  */
 
 /**
+ * Packet parse parameters
+ */
+typedef struct odp_packet_parse_param_t {
+	/** Protocol header at parse starting point. Valid values for this
+	 *  field are: ODP_PROTO_ETH, ODP_PROTO_IPV4, ODP_PROTO_IPV6. */
+	odp_proto_t proto;
+
+	/** Continue parsing until this layer. Must be the same or higher
+	 *  layer than the layer of 'proto'. */
+	odp_proto_layer_t layer;
+
+	/** Flags to control payload data checks up to the selected parse
+	 *  layer. Checksum checking status can be queried for each packet with
+	 *  odp_packet_l3_chksum_status() and odp_packet_l4_chksum_status().
+	 */
+	union {
+		struct {
+			/** Check IPv4 header checksum */
+			uint32_t ipv4_chksum   : 1;
+
+			/** Check UDP checksum */
+			uint32_t udp_chksum    : 1;
+
+			/** Check TCP checksum */
+			uint32_t tcp_chksum    : 1;
+
+		} check;
+
+		/** All check bits. This can be used to set/clear all flags. */
+		uint32_t all_check;
+	};
+
+} odp_packet_parse_param_t;
+
```

Comment:
Now that we have a `param` struct we should have an `odp_packet_parse_param_init()` API as well for completeness.

Reply:
Param init on fast path is a bit overkill. However, application could call it once and store the result. So, it could avoid extra function call for every packet. So, I'll add it.

Reply:
@psavol `_param_init` seems like an overkill in this case. From my point of view, `param_init` are good for object creation params (where we might not know platform-optimized defaults) but are an overkill for operation functions (we do not have `param_init` for ipsec, crypto, etc operational params).

Reply:
Then I don't add it now. We can add param init also later if param struct is extended with other things than check flags. Todays spec defines already that all flags should be set to zero.

 * @param param   Parse parameters. Proto and layer fields must be set. Clear
 *                all check bits that are not used.